modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -253,25 +247,21 @@ def get_completed_phase_number_from_phase_details(details)
         end
 
         def get_bgs_phase_completed_dates(data)
-          phase_dates = {}
-
-          if data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status).is_a?(Array)
-            max_completed_phase = 0
-            data[:benefit_claim_details_dto][:bnft_claim_lc_status].each_with_index do |lc, i|
-              completed_phase_number = get_completed_phase_number_from_phase_details(lc)
-              if i.zero?
-                max_completed_phase = completed_phase_number
-                phase_dates["phase#{completed_phase_number}CompleteDate"] = date_present(lc[:phase_chngd_dt])
-              elsif completed_phase_number.present? && completed_phase_number < max_completed_phase
-                phase_dates["phase#{completed_phase_number}CompleteDate"] = date_present(lc[:phase_chngd_dt])
+          lc_status_array =
+            [data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)].flatten.compact
+          # compact eliminates nils
+          max_completed_phase = lc_status_array.first[:phase_type_change_ind].split('').first
```

Comment:
Should we use get_completed_phase_number_from_phase_details in place of this assignment to max_completed_phase?

Reply:
That was my instinct too, then I wanted to replace the logic in that method, and that broke the third instance of that method. SO, I decided maybe that refactor should be another ticket since this one has been stuck in limbo for the entire week.